microcosm_elasticsearch/fields.py

```diff
@@ -0,0 +1,35 @@
+from enum import Enum
+from elasticsearch_dsl import Keyword
+from elasticsearch_dsl.exceptions import ValidationException
+
+
+class EnumField(Keyword):
```

Comment:
Can we make it so that they can customize what type this is based on? Just in case they want to use Text instead? Or in that case do we expect them to not use EnumField?

Reply:
@saivickna the way I see it, an `EnumField` is meant to only be searched using term filtering; otherwise it seems to defeat the purpose of using an enum. WDYT?

Reply:
Yep thats fair. We'd probably want less restrictions on Text fields.